src/brsTypes/components/RoDeviceInfo.ts

```diff
@@ -0,0 +1,687 @@
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            let locale = process.env.LOCALE;
+            return locale ? new BrsString(locale) : BrsInvalid.Instance;
+        },
+    });
+
+    private getCountryCode = new Callable("getCountryCode", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            let countryCode = process.env.LOCALE;
+            return countryCode ? new BrsString(countryCode) : BrsInvalid.Instance;
+        },
+    });
+
+    private getPreferredCaptionLanguage = new Callable("getPreferredCaptionLanguage", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("eng");
+        },
+    });
+
+    private timeSinceLastKeyPress = new Callable("timeSinceLastKeyPress", {
+        signature: {
+            args: [],
+            returns: ValueKind.Int32,
+        },
+        impl: (_interpreter) => {
+            return new Int32(360);
+        },
+    });
+
+    private getDrmInfo = new Callable("getDrmInfo", {
+        signature: {
+            args: [],
+            returns: ValueKind.Object,
+        },
+        impl: (_interpreter) => {
+            let result = new Array<AAMember>();
+            result.push({ name: new BrsString("playready"), value: new BrsString("tee;ss") });
+
+            return new RoAssociativeArray(result);
+        },
+    });
+
+    private getDrmInfoEx = new Callable("getDrmInfoEx", {
+        signature: {
+            args: [],
+            returns: ValueKind.Object,
+        },
+        impl: (_interpreter) => {
+            let result = new Array<AAMember>();
+            let subresult = new Array<AAMember>();
+            subresult.push({ name: new BrsString("multikey"), value: BrsBoolean.False });
+            subresult.push({ name: new BrsString("securestop"), value: BrsBoolean.True });
+            subresult.push({ name: new BrsString("tee"), value: BrsBoolean.False });
+            subresult.push({ name: new BrsString("version"), value: new Int32(2) });
+
+            result.push({
+                name: new BrsString("playready"),
+                value: new RoAssociativeArray(subresult),
+            });
+
+            return new RoAssociativeArray(result);
+        },
+    });
+
+    private getCaptionsMode = new Callable("getCaptionsMode", {
+        signature: {
+            args: [],
+            returns: ValueKind.String,
+        },
+        impl: (_interpreter) => {
+            return new BrsString("On");
+        },
+    });
+
+    private setCaptionsMode = new Callable("setCaptionsMode", {
+        signature: {
+            args: [new StdlibArgument("mode", ValueKind.String)],
+            returns: ValueKind.Boolean,
+        },
+        impl: (_interpreter) => {
+            return BrsBoolean.True;
```

Comment:
We want `getCaptionsMode` to return this value I believe. So let's initialize a private variable in this class to keep track of captions mode, something like `this.captionsMode`. Also, can you check to see what the behavior is if the user tries to set captions mode to a string that's not one of the 4 supported options? I suspect it's a no-op, but regardless we should emulate it.